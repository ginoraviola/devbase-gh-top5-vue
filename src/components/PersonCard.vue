<script lang="ts">
import type { PropType } from "vue";
import { defineComponent } from "vue";

interface IUser {
  name: string;
  location: string;
  avatar_url: string;
}

export default defineComponent({
  props: {
    user: {
      type: Object as PropType<IUser>,
      required: true,
    },
  },
});
</script>

<template>
  <div class="flex my-8 lg:px-4">
    <img
      :src="user.avatar_url"
      :alt="user.name"
      class="rounded-full w-50px h-50px lg:w-75px lg:h-75px mr-5"
    />
    <p class="flex flex-col text-start justify-center">
      <span class="text-lg lg:text-xl font-semibold" v-text="user.name"></span>
      <span class="text-md text-gray-400 lg:font-light" v-text="user.location"></span>
    </p>
  </div>
</template>
