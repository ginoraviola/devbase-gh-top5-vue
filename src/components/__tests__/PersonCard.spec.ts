import { describe, it, expect } from "vitest";

import { mount } from "@vue/test-utils";
import PersonCard from "../PersonCard.vue";

describe("Person", () => {
  it("renders properly", () => {
    const wrapper = mount(PersonCard, {
      props: {
        user: {
          name: "John Doe",
          location: "San Francisco",
          avatar_url: "https://avatars.githubusercontent.com/u/2829600?v=4",
        },
      },
    });
    expect(wrapper.text()).toContain("John Doe");
    expect(wrapper.text()).toContain("San Francisco");
    expect(wrapper).toMatchSnapshot();
  });
});
