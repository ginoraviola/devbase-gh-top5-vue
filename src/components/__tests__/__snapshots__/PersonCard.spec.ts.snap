// Vitest Snapshot v1

exports[`Person > renders properly 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="flex my-8 lg:px-4"
      >
        <img
          alt="John Doe"
          class="rounded-full w-50px h-50px lg:w-75px lg:h-75px mr-5"
          src="https://avatars.githubusercontent.com/u/2829600?v=4"
        />
        <p
          class="flex flex-col text-start justify-center"
        >
          <span
            class="text-lg lg:text-xl font-semibold"
          >
            John Doe
          </span>
          <span
            class="text-md text-gray-400 lg:font-light"
          >
            San Francisco
          </span>
        </p>
      </div>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {},
      "da": null,
      "data": {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {},
      "refs": {
        "VTU_COMPONENT": {
          "hasOwnProperty": [Function],
          "user": {
            "avatar_url": "https://avatars.githubusercontent.com/u/2829600?v=4",
            "location": "San Francisco",
            "name": "John Doe",
          },
        },
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": [],
        "effects": [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": {},
      "slots": {},
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "_isMockFunction": 0,
            "constructor": 0,
            "hasOwnProperty": 4,
            "toJSON": 0,
            "user": 3,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "hasOwnProperty": [Function],
            "user": {
              "avatar_url": "https://avatars.githubusercontent.com/u/2829600?v=4",
              "location": "San Francisco",
              "name": "John Doe",
            },
          },
          "da": null,
          "data": {},
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {
            "user": {
              "avatar_url": "https://avatars.githubusercontent.com/u/2829600?v=4",
              "location": "San Francisco",
              "name": "John Doe",
            },
          },
          "propsDefaults": {},
          "propsOptions": [
            {
              "user": {
                "0": false,
                "1": true,
                "required": true,
                "type": [Function],
              },
            },
            [],
          ],
          "provides": {},
          "proxy": {
            "hasOwnProperty": [Function],
            "user": {
              "avatar_url": "https://avatars.githubusercontent.com/u/2829600?v=4",
              "location": "San Francisco",
              "name": "John Doe",
            },
          },
          "refs": {},
          "render": [Function],
          "renderCache": [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": [],
            "effects": [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": null,
          "setupState": {},
          "slots": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": [
                  "src",
                  "alt",
                ],
                "el": <img
                  alt="John Doe"
                  class="rounded-full w-50px h-50px lg:w-75px lg:h-75px mr-5"
                  src="https://avatars.githubusercontent.com/u/2829600?v=4"
                />,
                "key": null,
                "patchFlag": 8,
                "props": {
                  "alt": "John Doe",
                  "class": "rounded-full w-50px h-50px lg:w-75px lg:h-75px mr-5",
                  "src": "https://avatars.githubusercontent.com/u/2829600?v=4",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "img",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": [
                      "textContent",
                    ],
                    "el": <span
                      class="text-lg lg:text-xl font-semibold"
                    >
                      John Doe
                    </span>,
                    "key": null,
                    "patchFlag": 8,
                    "props": {
                      "class": "text-lg lg:text-xl font-semibold",
                      "textContent": "John Doe",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "span",
                  },
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": [
                      "textContent",
                    ],
                    "el": <span
                      class="text-md text-gray-400 lg:font-light"
                    >
                      San Francisco
                    </span>,
                    "key": null,
                    "patchFlag": 8,
                    "props": {
                      "class": "text-md text-gray-400 lg:font-light",
                      "textContent": "San Francisco",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "span",
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <p
                  class="flex flex-col text-start justify-center"
                >
                  <span
                    class="text-lg lg:text-xl font-semibold"
                  >
                    John Doe
                  </span>
                  <span
                    class="text-md text-gray-400 lg:font-light"
                  >
                    San Francisco
                  </span>
                </p>,
                "key": null,
                "patchFlag": 0,
                "props": {
                  "class": "flex flex-col text-start justify-center",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "p",
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": [
                  "src",
                  "alt",
                ],
                "el": <img
                  alt="John Doe"
                  class="rounded-full w-50px h-50px lg:w-75px lg:h-75px mr-5"
                  src="https://avatars.githubusercontent.com/u/2829600?v=4"
                />,
                "key": null,
                "patchFlag": 8,
                "props": {
                  "alt": "John Doe",
                  "class": "rounded-full w-50px h-50px lg:w-75px lg:h-75px mr-5",
                  "src": "https://avatars.githubusercontent.com/u/2829600?v=4",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "img",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": [
                  "textContent",
                ],
                "el": <span
                  class="text-lg lg:text-xl font-semibold"
                >
                  John Doe
                </span>,
                "key": null,
                "patchFlag": 8,
                "props": {
                  "class": "text-lg lg:text-xl font-semibold",
                  "textContent": "John Doe",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "span",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": [
                  "textContent",
                ],
                "el": <span
                  class="text-md text-gray-400 lg:font-light"
                >
                  San Francisco
                </span>,
                "key": null,
                "patchFlag": 8,
                "props": {
                  "class": "text-md text-gray-400 lg:font-light",
                  "textContent": "San Francisco",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "span",
              },
            ],
            "dynamicProps": null,
            "el": <div
              class="flex my-8 lg:px-4"
            >
              <img
                alt="John Doe"
                class="rounded-full w-50px h-50px lg:w-75px lg:h-75px mr-5"
                src="https://avatars.githubusercontent.com/u/2829600?v=4"
              />
              <p
                class="flex flex-col text-start justify-center"
              >
                <span
                  class="text-lg lg:text-xl font-semibold"
                >
                  John Doe
                </span>
                <span
                  class="text-md text-gray-400 lg:font-light"
                >
                  San Francisco
                </span>
              </p>
            </div>,
            "key": null,
            "patchFlag": 0,
            "props": {
              "class": "flex my-8 lg:px-4",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "/Users/ginoraviola/Workspace/challenges/devbase-gh-top5/src/components/PersonCard.vue",
            "components": {},
            "props": {
              "user": {
                "0": false,
                "1": true,
                "required": true,
                "type": [Function],
              },
            },
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="flex my-8 lg:px-4"
        >
          <img
            alt="John Doe"
            class="rounded-full w-50px h-50px lg:w-75px lg:h-75px mr-5"
            src="https://avatars.githubusercontent.com/u/2829600?v=4"
          />
          <p
            class="flex flex-col text-start justify-center"
          >
            <span
              class="text-lg lg:text-xl font-semibold"
            >
              John Doe
            </span>
            <span
              class="text-md text-gray-400 lg:font-light"
            >
              San Francisco
            </span>
          </p>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "ref": "VTU_COMPONENT",
          "user": {
            "avatar_url": "https://avatars.githubusercontent.com/u/2829600?v=4",
            "location": "San Francisco",
            "name": "John Doe",
          },
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "__file": "/Users/ginoraviola/Workspace/challenges/devbase-gh-top5/src/components/PersonCard.vue",
          "components": {},
          "props": {
            "user": {
              "0": false,
              "1": true,
              "required": true,
              "type": [Function],
            },
          },
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="flex my-8 lg:px-4"
        >
          <img
            alt="John Doe"
            class="rounded-full w-50px h-50px lg:w-75px lg:h-75px mr-5"
            src="https://avatars.githubusercontent.com/u/2829600?v=4"
          />
          <p
            class="flex flex-col text-start justify-center"
          >
            <span
              class="text-lg lg:text-xl font-semibold"
            >
              John Doe
            </span>
            <span
              class="text-md text-gray-400 lg:font-light"
            >
              San Francisco
            </span>
          </p>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.37",
  },
  "__setProps": [Function],
  "componentVM": {
    "hasOwnProperty": [Function],
    "user": {
      "avatar_url": "https://avatars.githubusercontent.com/u/2829600?v=4",
      "location": "San Francisco",
      "name": "John Doe",
    },
  },
  "isDisabled": [Function],
  "rootVM": {},
  "wrapperElement": <div
    class="flex my-8 lg:px-4"
  >
    <img
      alt="John Doe"
      class="rounded-full w-50px h-50px lg:w-75px lg:h-75px mr-5"
      src="https://avatars.githubusercontent.com/u/2829600?v=4"
    />
    <p
      class="flex flex-col text-start justify-center"
    >
      <span
        class="text-lg lg:text-xl font-semibold"
      >
        John Doe
      </span>
      <span
        class="text-md text-gray-400 lg:font-light"
      >
        San Francisco
      </span>
    </p>
  </div>,
}
`;
